src/components/TextInput/TextInput.test.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import {TextInputEvent} from './types';
+import {TextInput} from './TextInput';
+import {TextAreaControl} from './TextAreaControl/TextAreaControl';
+import {InputControl} from './InputControl/InputControl';
+
+describe('TextInput', () => {
+    it('should render text area component when component receives multiline props', () => {
+        const wrapper = shallow(<TextInput multiline />);
+
+        expect(wrapper.find(TextAreaControl).exists()).toBeTruthy();
+    });
+
+    it("shouldn render text input component when component doesn't receive multiline props", () => {
+        const wrapper = shallow(<TextInput />);
+
+        expect(wrapper.find(InputControl).exists()).toBeTruthy();
+    });
+
+    it('should render error message in input when component receives error props', () => {
+        const wrapper = shallow(<TextInput error="FAKE_ERROR" />);
+
+        expect(wrapper.find('[data-qa="yc-text-input__error"]').exists()).toBeTruthy();
+    });
+
+    it("shouldn't render error message in input when component doesn't receive error props", () => {
+        const wrapper = shallow(<TextInput />);
+
+        expect(wrapper.find('[data-qa="yc-text-input__error"]').exists()).toBeFalsy();
+    });
+
+    it('should render clear button when component receives hasClear props', () => {
+        const wrapper = shallow(<TextInput hasClear />);
+
+        expect(wrapper.find('[data-qa="yc-text-input__clear"]').exists()).toBeTruthy();
+    });
+
+    it("shouldn't render clear button when component doesn't receive hasClear props", () => {
+        const wrapper = shallow(<TextInput />);
+
+        expect(wrapper.find('[data-qa="yc-text-input__clear"]').exists()).toBeFalsy();
+    });
+
+    it('should call onChange with event function when input changes value', () => {
+        const FAKE_ON_CHANGE_FN = jest.fn();
+        const FAKE_EVENT = {target: {}} as TextInputEvent;
+
+        const wrapper = shallow(<TextInput onChange={FAKE_ON_CHANGE_FN} />);
+        wrapper.find(InputControl).props().onChange?.(FAKE_EVENT);
```

Comment:
Better not trigger function itself, but change value of the input via `simulate('change')`

Reply:
It is not impossible. Here is unit tests and we can't simulate event inside other components